src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,877 @@
+            
+            @Override
+            public void setSelectedValuesList(List<Object> selectedValuesList, int[] selectedIndices) {
+                selectedLanguagesIndexes = selectedIndices;
+                updateLanguagesLabel(selectedValuesList);
+            }
+        });
+        selectLanguagessFrame.setVisible(true);
+    }
+ 
+
+    protected void showProblemList() {
+        
+        Problem[] problems = getContest().getProblems();
+        
+        if (problems.length == 0) {
+            showMessage("No problems defined in contest");
+            return;
+        }
+    
+        JListFrame selectProblemsFrame = new JListFrame("Select Problems ", problems, selectedProblemsIndexes, new ISelectedListsSetter() {
+            
+            @Override
+            public void setSelectedValuesList(List<Object> selectedValuesList, int[] selectedIndices) {
+                selectedProblemsIndexes = selectedIndices;
+                updateProblemsLabel(selectedValuesList);
+            }
+        });
+        selectProblemsFrame.setVisible(true);
+    }
+
+    protected void updateCDPDirAndFields(String cdpConfigDir) {
+	    try {
+            File configFile = new File(cdpConfigDir);
+            String cdpRootDirectory = configFile.getAbsolutePath();
+//            if (configFile.getAbsolutePath().endsWith(IContestLoader.CONFIG_DIRNAME)) {
+//                cdpRootDirectory = configFile.getParent();
+//            }
+	        
+	        cdpTextField.setText(cdpRootDirectory);
+	        updateClientCDPPath(cdpRootDirectory);
+	        updateNosLabel(null);
+        } catch (Exception e) {
+            log.log(Level.WARNING, "Problem updating CDP Dir", e);
+        }
+    }
+
+    /**
+	 * Dialog with question mark ison for info on selecting judgement types.
+	 * @return
+	 */
+	private JLabel getWhatsThisOne() {
+		
+		String[] messageLines = {
+				"Selecting judgement types", //
+				"", //
+				"Use Select button to select judgement types", //
+		};
+		
+		return FrameUtilities.getWhatsThisLabel("Selecting judgement samples", messageLines);
+	}
+	
+	/**
+	 * Repopulate submissionSolutionList.
+	 * 
+	 * @return 
+	 */
+	public SubmissionSolutionList getSubmissionSolutionList() {
+		String cdpPath = cdpTextField.getText();
+		
+		SubmissionSolutionList list = new SubmissionSolutionList(new File(cdpPath));
+		Collections.reverse(list);
+		submissionSolutionList = list;
+		
+		return submissionSolutionList;
+	}
+
+	protected void showJudgementTypeList() {
+		
+		if (getSubmissionSolutionList() == null || getSubmissionSolutionList().size() == 0) {
+			showMessage("No submission judgement types found in dir: "+cdpTextField.getText());
+			return;
+		}
+		
+		SubmissionSampleLocation[] listData = toArray(getSubmissionSolutionList());
+	
+		JListFrame selectNoSolutionsFrame = new JListFrame("Select No/Failed Judges solutions", listData, selectedJudgingTypeIndexes, new ISelectedListsSetter() {
```

Comment:
This constructs a new `JListFrame` every time `showJudgementTypeList()` is called.  `JListFrame` derives from ("is-a") JFrame.  The default Close operation on a JFrame is "hide the window".  This means there are going to be many hidden windows floating around, never used.   Either the default close operation on this instance should be changed, or the `JListFrame` class itself should be changed so that it does a "dispose on close".

Note: the same flaw exist in `showLanguageList()`, above.

Reply:
I consider this an approval showstopper.